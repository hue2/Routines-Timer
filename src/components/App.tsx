import * as React from 'react';
import '../styles/App.css';
import '../styles/Options.css';
import Clock from './Clock';
import ClockOptions from './ClockOptions';
import Spinner from './Spinner';
import 'font-awesome/css/font-awesome.min.css';
import { ITimeInfo, ITimeBreak, ITimeRepeat, AppProps, AppState } from './TimeType';

const defaultTime: ITimeInfo = {
  hour: 0,
  minute: 0,
  second: 0,
  tempHour: 0,
  tempMinute: 0,
  tempSecond: 0,
}

const defaultBreak: ITimeBreak = {
  minuteBreak: 0,
  secondBreak: 0,
  tempMinuteBreak: 0,
  tempSecondBreak: 0,
}

const defaultRepeat: ITimeRepeat = {
  repeat: 0,
  tempRepeat: 0,
}

export default class App extends React.Component<AppProps, AppState> {
  constructor(props: AppProps) {
    super(props);
    this.state = { show: false, value: 0, 
        time: {...defaultTime},
        timeBreak: {...defaultBreak},
        repeats: {...defaultRepeat}, 
        isBreak: false,           
        tempBreak: {...defaultBreak},
        isPaused: false, 
        showOptions: false, 
        tempRepeats: 0, 
        isStart: false,
        notify: false,
      }
  }
 
  handleNonIntInput = (input: any) => {
    if (!isNaN(input)) {
      return input.length < 2 ? +("0" + input) : +input ;
    }
    return 0;
  }

  handleOpenNav = () => this.setState({ show: true });
  handleCloseNav = () => this.setState({ show: false });
  
  handleInputChange = (event: any) => {
    let currentState = {...this.state[event.target.dataset.state]};
    event.target.classList.forEach((value, key, listObj) => {
      currentState[value] = this.handleNonIntInput(event.target.value);
    });  
    this.setState({ [event.target.dataset.state]: currentState } as Pick<AppState, keyof AppState>);
  }

  handleStartTick = () => {
    if (!this.state.isPaused) {
      this.handleTimerTick();
    }
  }

  handleClearTimer = () => {
      //@ts-ignore
      clearInterval(this.interval);
  }

  handlePause = () => {
      //if it's currently paused, then start
      if (this.state.isPaused == true) {
        this.handleTimerTick();
        this.handleStartTimer();
      }

      //if it's currently start, then pause
      else {
        this.setState({ isPaused: true, isStart: false });
        this.handleClearTimer();
      }
  }

  handleTimerTick = () => {
    if (this.state.value === -1) {
      this.handleClearTimer();
      this.handleStartTimer();
    } else if (!this.state.isBreak) {
      this.setState({ notify: false });
      this.parseTime();    
    }    
    else if (this.state.isBreak) {
      this.setState({ notify: false });
      this.parseBreakTime();     
    }
  }

  handleReset = () => {
    this.setState({ 
        time: {...defaultTime},
        timeBreak: {...defaultBreak},
        repeats: {...defaultRepeat}, 
        isBreak: false,           
        tempBreak: {...defaultBreak},
        isPaused: false, 
        showOptions: false, 
        tempRepeats: 0, 
        isStart: false,
        show: false, 
        value: 0,
        notify: false,
    });
    this.handleClearTimer();
  }

  handleRestart = () => {
    this.handleClearTimer();

    this.setState({
      time: {...this.state.time, 
        tempHour: this.state.time.hour,
        tempMinute: this.state.time.minute,
        tempSecond: this.state.time.second
      },
      timeBreak: {...this.state.timeBreak,
        tempMinuteBreak: this.state.timeBreak.minuteBreak,
        tempSecondBreak: this.state.timeBreak.secondBreak
      },
      repeats: {...this.state.repeats,
        tempRepeat: this.state.repeats.repeat
      },
      isPaused: false,
      isBreak: false,
      showOptions: false,
      isStart: false,
      value: 0,
      notify: false,
    });    
  }

  checkConditions = () => ({
      isBreak: this.state.time.tempSecond < 1 && this.state.time.tempMinute < 1 && this.state.time.tempHour < 1 && !this.state.isBreak,
      hasTime: this.state.time.tempSecond > 0 || this.state.time.tempMinute > 0 || this.state.time.tempHour > 0,
      hasBreaks: this.state.timeBreak.secondBreak > 0 || this.state.timeBreak.minuteBreak > 0,
      hasRepeats: this.state.repeats.tempRepeat > 0 && this.state.timeBreak.tempSecondBreak < 1 && this.state.timeBreak.tempMinuteBreak < 1    
  });


  parseBreakTime = () => {
    let totalSeconds = this.state.value < 0 && (this.state.timeBreak.tempSecondBreak >= 1 || this.state.timeBreak.tempMinuteBreak >= 1) ? 
      +this.state.timeBreak.tempSecondBreak + (this.state.timeBreak.tempMinuteBreak * 60) : this.state.value;
    let seconds = totalSeconds % 60;
    let minAndHourInMinutes = (totalSeconds - seconds) / 60;
    var minutes = minAndHourInMinutes % 60;
  
    let timeBreak = {...this.state.timeBreak, 
      tempMinuteBreak: this.handleNumberFormat(minutes), 
      tempSecondBreak: this.handleNumberFormat(seconds) }
    this.setState({ value: totalSeconds - 1, timeBreak, isBreak: true, isStart: false });
  }

  parseTime = () => {
    let totalSeconds = this.state.value <= 0 && (this.state.time.tempSecond > 1 || 
      this.state.time.tempMinute >= 1 || this.state.time.tempHour >= 1) ? 
      +this.state.time.tempSecond + (this.state.time.tempMinute * 60) + (this.state.time.tempHour * 3600) : this.state.value;
    let seconds = totalSeconds % 60;
    let minAndHourInMinutes = (totalSeconds - seconds) / 60;
    var minutes = minAndHourInMinutes % 60;
    let hour = (minAndHourInMinutes - minutes) / 60;  
  
    let time = {...this.state.time, 
            tempMinute: this.handleNumberFormat(minutes), 
            tempSecond: this.handleNumberFormat(seconds), 
            tempHour: this.handleNumberFormat(hour) };
    this.setState({  value: totalSeconds - 1, time, isStart: true });
  }


  handleNumberFormat = (time: number) => {
      return time.toString().length < 2 ? +("0" + time) : +time;
  }

  handleTimeConvert = () => {
        let currentStatus = this.checkConditions();
        //all the time has been exhausted, start break if there's any
        if (currentStatus.isBreak && currentStatus.hasBreaks) {
            this.parseBreakTime();
            this.setState({ notify: true });
        }
        else {
          this.setState({ isBreak: false });
          if (!currentStatus.hasTime) {
            this.setState({ isStart: false, notify: currentStatus.hasRepeats ? false : true  });
          }
        }

        if (currentStatus.hasTime) {
          this.parseTime();
        }

        //if there's no time or no break left, check for repeats
        else if (currentStatus.hasRepeats) {
          this.setState({ 
            repeats: {...this.state.repeats, 
              tempRepeat: this.state.repeats.tempRepeat - 1 },
            time: {...this.state.time, 
              tempMinute: this.state.time.minute, 
              tempSecond: this.state.time.second, 
              tempHour: this.state.time.hour},
            timeBreak: {...this.state.timeBreak, 
              tempMinuteBreak: this.state.timeBreak.minuteBreak, 
              tempSecondBreak: this.state.timeBreak.secondBreak },
            isBreak: false
          });
    }
  }

  handleStartTimer = () => { 
      this.handleCloseNav();

      //this check will allow minutes and hours to be converted to seconds for countdown
      if (this.state.value < 1) {
          this.handleTimeConvert();
      }
      this.setState({ isPaused: false, showOptions: true })

      if (this.state.time.tempHour > 0 || this.state.time.tempMinute > 0 || this.state.time.tempSecond > 0 || 
        this.state.repeats.tempRepeat > 0 || this.state.timeBreak.tempSecondBreak > 0 || this.state.timeBreak.tempMinuteBreak > 0) {    
          //@ts-ignore
        this.interval = setInterval(this.handleStartTick, 1000);      
      }
  }

  render() {
    return (
      <div id="main">
          <div id="clock-background">

          </div>
          <div className={this.state.isStart ? "App-start" : "App"}>
            <Spinner isStart={this.state.isStart} />
            <Clock hour={this.state.time.tempHour} 
              minute={this.state.time.tempMinute}
              seconds={this.state.time.tempSecond} 
              minuteBreak={this.state.timeBreak.tempMinuteBreak} 
              secondBreak={this.state.timeBreak.tempSecondBreak} 
              repeats={this.state.repeats.repeat} 
              handleStartTimer={this.handleStartTimer}
              isBreak={this.state.isBreak}
              handlePause={this.handlePause}
              isPaused={this.state.isPaused}
              showOptions={this.state.showOptions}
              onReset={this.handleReset}
              onRestart={this.handleRestart}
              notify={this.state.notify}
              remainingRepeats={this.state.repeats.tempRepeat}
            />
            <ClockOptions 
              hour={this.state.time.hour} 
              minute={this.state.time.minute}
              seconds={this.state.time.second} 
              minuteBreak={this.state.timeBreak.minuteBreak} 
              secondBreak={this.state.timeBreak.tempSecondBreak} 
              repeats={this.state.repeats.repeat} 
              navOpen={this.state.show} 
              navClose={this.handleCloseNav}
              handleChange={this.handleInputChange}
            />
            <div id="handleOpenNav" data-testid="settings-btn" onClick={this.handleOpenNav} className={this.state.showOptions ? "hidden" : ""}><i className="fa fa-gear"></i> 
              Settings
            </div>
        </div>
        <div id="footer">
          Image by <a href="https://pixabay.com/users/moinzon-2433302/?utm_source=link-attribution&amp;utm_medium=referral&amp;utm_campaign=image&amp;utm_content=1412683" className="footer-link">Michi S</a> from 
             &nbsp;<a href="https://pixabay.com/?utm_source=link-attribution&amp;utm_medium=referral&amp;utm_campaign=image&amp;utm_content=1412683" className="footer-link">Pixabay</a>
        </div>
      </div>
    );
  }
}