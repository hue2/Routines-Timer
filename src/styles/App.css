.gradient-background, html {
  background: #141E30;
  /* fallback for old browsers */
  background: -webkit-linear-gradient(to top, #243B55, #141E30);
  /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(to top, #243B55, #141E30);
  /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */ }

body {
  color: white; }

.hidden {
  display: none; }

.inline-block, .timer-btn, .disabled-btn, .input-time {
  display: inline-block; }

.App {
  padding-top: 30vh;
  text-align: center;
  height: 100vh; }

.App-start {
  margin-top: 30vh;
  text-align: center; }

.timer-btn, .disabled-btn {
  border-radius: 20px;
  color: white;
  border: none;
  padding: 20px 60px;
  margin: 15px 30px;
  cursor: pointer;
  outline: none;
  font-size: 20pt;
  background: none; }
  .timer-btn:active, .disabled-btn:active {
    top: 3px;
    left: 2px; }
  @media screen and (max-width: 500px) {
    .timer-btn, .disabled-btn {
      font-size: 12pt;
      margin-top: -10%; } }

.primary-btn {
  color: #93DFB8; }
  .primary-btn:hover {
    cursor: pointer;
    text-decoration: underline; }

.orange-btn {
  color: orange; }
  .orange-btn:hover {
    cursor: pointer;
    text-decoration: underline; }

.red-btn {
  color: #FF355E; }
  .red-btn:hover {
    cursor: pointer;
    text-decoration: underline; }

.gray-btn {
  background: none;
  color: white; }
  .gray-btn:hover {
    cursor: pointer;
    text-decoration: underline; }

#handleOpenNav {
  margin-top: 20px; }
  #handleOpenNav:hover {
    cursor: pointer;
    color: #3493ce; }

body, html {
  margin: 0;
  padding: 0; }

#background {
  background: url("lake.jpg");
  background-repeat: no-repeat; }

#modal {
  width: 100px;
  height: 100px;
  border: 1px; }

.time {
  font-size: 70px;
  margin-top: -100px;
  padding-top: 100px; }
  @media screen and (max-width: 500px) {
    .time {
      font-size: 30pt; } }

.padding-200px {
  padding-bottom: 200px; }

#OptionsModal {
  margin-left: 20%; }

input {
  height: 50px;
  float: left;
  margin-top: 40px;
  font-size: 30px;
  margin-left: 5px;
  border-bottom: 1px solid white;
  border-left: none;
  border-right: none;
  border-top: none;
  width: 60px;
  padding-bottom: 0px;
  text-align: center;
  background: none;
  color: white; }

.dot-separator {
  background: none;
  border: none;
  width: 5px;
  font-weight: 800; }

input:focus {
  outline-width: 0;
  caret-color: white; }

.fa-gear {
  margin-right: 5px; }

.disabled-btn {
  color: #c5c8cc;
  cursor: not-allowed !important; }

.small-text {
  font-size: 16pt; }

html {
  width: 100%;
  height: 100%;
  background-color: #080808; }

ul {
  height: 100%;
  width: 100%;
  display: block;
  margin: 0 auto;
  position: fixed;
  top: -20vh;
  left: -4vw;
  pointer-events: none; }
  @media screen and (max-width: 1200px) {
    ul {
      left: -6vw; } }
  @media screen and (max-width: 768px) {
    ul {
      left: -8vw; } }

li {
  position: absolute;
  left: 50%;
  top: 50%;
  display: block;
  background: transparent;
  border: 3px solid #17f6fb;
  border-left-color: transparent;
  border-right-color: transparent;
  border-radius: 500px;
  transition: all 0.5s ease; }

li:first-child {
  margin-left: -130px;
  margin-top: -130px;
  z-index: 2;
  width: 340px;
  height: 340px;
  border-width: 5px;
  animation: spinBG 5s infinite linear; }
  @media screen and (max-width: 500px) {
    li:first-child {
      width: 220px;
      height: 220px;
      margin-left: -100px;
      margin-top: -100px; } }

li:nth-child(2) {
  margin-left: -137px;
  margin-top: -137px;
  z-index: 1;
  width: 370px;
  height: 370px;
  border-width: 1px;
  border-style: solid;
  box-shadow: 0px 0px 20px rgba(23, 246, 251, 0.5);
  animation: spinBG2 2s infinite linear; }
  @media screen and (max-width: 500px) {
    li:nth-child(2) {
      width: 270px;
      height: 270px;
      margin-left: -120px;
      margin-top: -120px; } }

li:nth-child(3) {
  margin-left: -150px;
  margin-top: -150px;
  z-index: 1;
  width: 396px;
  height: 396px;
  border-width: 2px;
  box-shadow: inset 0px 0px 25px rgba(23, 246, 251, 0.25);
  animation: spinBG 12s infinite linear; }
  @media screen and (max-width: 500px) {
    li:nth-child(3) {
      width: 305px;
      height: 305px;
      margin-left: -140px;
      margin-top: -140px; } }

li:nth-child(4) {
  margin-left: -170px;
  margin-top: -170px;
  z-index: 1;
  width: 430px;
  height: 430px;
  border-width: 5px;
  border-style: solid;
  box-shadow: inset 0px 0px 25px #17f6fb;
  animation: spinBG3 8s infinite linear; }
  @media screen and (max-width: 500px) {
    li:nth-child(4) {
      width: 325px;
      height: 325px;
      margin-left: -150px;
      margin-top: -150px; } }

/*-------------------------------------------
 Animations
-------------------------------------------*/
@keyframes pulseGlow {
  0% {
    text-shadow: 0px 0px 20px rgba(23, 246, 251, 0.75); }
  50% {
    text-shadow: 0px 0px 40px rgba(23, 246, 251, 0.5); }
  100% {
    text-shadow: 0px 0px 20px rgba(23, 246, 251, 0.75); } }

@keyframes pulseShadow {
  0% {
    box-shadow: 0px 0px 30px rgba(23, 246, 251, 0.25); }
  50% {
    box-shadow: 0px 0px 30px rgba(23, 246, 251, 0.75); }
  100% {
    box-shadow: 0px 0px 30px rgba(23, 246, 251, 0.25); } }

@keyframes spinBG {
  0% {
    transform: rotate(0deg); }
  100% {
    transform: rotate(360deg); } }

@keyframes spinBG2 {
  0% {
    transform: rotate(360deg);
    box-shadow: 0px 0px 1px rgba(23, 246, 251, 0.5); }
  50% {
    transform: rotate(180deg);
    box-shadow: 0px 0px 20px rgba(23, 246, 251, 0.5); }
  100% {
    transform: rotate(0deg);
    box-shadow: 0px 0px 1px rgba(23, 246, 251, 0.5); } }

@keyframes spinBG3 {
  0% {
    transform: rotate(180deg);
    box-shadow: 0px 0px 1px rgba(23, 246, 251, 0.1); }
  50% {
    transform: rotate(0deg); }
  100% {
    transform: rotate(-180deg); } }

#break-countdown {
  margin-top: -200px;
  font-size: 70px;
  margin-top: -100px; }
