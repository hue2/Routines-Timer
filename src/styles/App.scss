$button-blue: #3493ce;
$hover-blue: #2b7daf;
$button-orange: #f4a742;
$hover-orange: #db963b;
$turquoise: #267871;
$gradient-blue: #136a8a;

.gradient-background {
  background: #141E30;  /* fallback for old browsers */
  background: -webkit-linear-gradient(to top, #243B55, #141E30);  /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(to top, #243B55, #141E30); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
}

html {
  @extend .gradient-background;
}

body {
  color: white;
}
.hidden {
  display: none;
}

.inline-block {
  display: inline-block;
}

.App {
  padding-top: 30vh;
  text-align: center;
}

.App-start {
  margin-top: 30vh;
  text-align: center;
}

.timer-btn {
  border-radius: 20px;
  color: white;
  border: none;
  padding: 20px 60px;
  @extend .inline-block;
  margin: 15px 30px;
  cursor: pointer;
  outline: none;
  font-size: 20pt;
  background: none;
  &:active {
    top: 3px;
    left: 2px;
  }

  @media screen and (max-width: 500px) {
    font-size: 12pt;
    margin-top: -10%;
  }
}

.primary-btn {
  color: #fff;
  text-align: center;
  -webkit-animation: glow 1s ease-in-out infinite alternate;
  -moz-animation: glow 1s ease-in-out infinite alternate;
  animation: glow 1s ease-in-out infinite alternate;
  &:hover {
    cursor: pointer;
    text-decoration: underline;
  }
}



.pause-btn {
  color: #FF756D;
  &:hover {
    
    cursor: pointer;
    text-decoration: underline;
  }
}

.reset-btn {
  color: #F9FFCB;
  &:hover {
    cursor: pointer;
    text-decoration: underline;
  }
}

.restart-btn {
  background: none;

  color: #FFCED0;
  &:hover {
    cursor: pointer;
    text-decoration: underline;
  }
}

#handleOpenNav {
  margin-top: 20px;
  font-size: 17px;
  font-weight: bold;
  &:hover {
    cursor: pointer;
    color: $button-blue;
  }
}

body, html{
  margin:0;
  padding:0;
}

#background {
  background: url('lake.jpg');
  background-repeat: no-repeat;
}
#modal{
  width: 100px;
  height: 100px;
  border: 1px;
}

.time {
  font-size: 70px;
  margin-top: -100px;
  padding-top: 80px;
  @media screen and (max-width: 500px) {
    font-size: 45px;
  }
}

.padding-200px {
  padding-bottom: 200px;
}

#OptionsModal {
  margin-left: 20%;
}

input {
  height: 50px;
  float: left;
  margin-top: 40px;
  font-size: 30px;
  margin-left: 5px;
  border-bottom: 1px solid white;
  border-left: none;
  border-right: none;
  border-top: none;
  width: 60px;
  padding-bottom: 0px;
  text-align: center;
  background: none;
  color: white;
}

.dot-separator{
  background: none;
  border: none;
  width: 5px;
  font-weight: 800;
}

input {
  &:focus {
    outline-width: 0;
    caret-color: white;
  }
}


.fa-gear {
  margin-right: 5px;
}

.input-time {
  @extend .inline-block;
}

.disabled-btn {
  @extend .timer-btn;
  color: #c5c8cc;
  cursor: not-allowed !important;
  display: none;
}

.small-text {
  font-size: 16pt;
}

html{  
  width: 100%;
	height: 100%;
	background-color: rgba(8,8,8,1.0);
}

.table {
  display: flex;
  justify-content: center;
}

ul{
  pointer-events: none;
}

li{
	position: absolute;
	// left: 50%;
	// top: 50%;
	display: block;
	background: transparent;
	border: 3px solid rgb(255,225,213);
	border-left-color: transparent;
	border-right-color: transparent;
	border-radius: 500px;
  transition: all 0.5s ease;
  margin-left: auto;
  margin-right: auto;
  left: 0;
  right: 0;
  
}

li:first-child{
	// margin-left: -130px;
	margin-top: -130px;
	z-index: 2;
	width: 340px;
	height: 340px;
	border-width: 5px;
  animation: spinBG 5s infinite linear;
  @media screen and (max-width: 500px) {
    width: 220px;
    height: 220px;
    margin-top: -100px;
  }
}

li:nth-child(2){
	// margin-left: -137px;
	margin-top: -137px;
	z-index: 1;
	width: 370px;
	height: 370px;
	border-width: 1px;
	border-style: solid;
	box-shadow: 0px 0px 20px rgba(217,185,218, 0.5);
  animation: spinBG2 2s infinite linear;
  @media screen and (max-width: 500px) {
    width: 270px;
    height: 270px;
    margin-top: -120px;
  }
}

li:nth-child(3){
	// margin-left: -150px;
	margin-top: -150px;
	z-index: 1;
	width: 396px;
	height: 396px;
	border-width: 2px;
	box-shadow: inset 0px 0px 25px rgba(217,185,218, 0.5);
  animation: spinBG 12s infinite linear;
  @media screen and (max-width: 500px) {
    width: 305px;
    height: 305px;
    margin-top: -140px;
  }
}

li:nth-child(4){
	// margin-left: -170px;
	margin-top: -170px;
	z-index: 1;
	width: 430px;
	height: 430px;
	border-width: 5px;
	border-style: solid;
	box-shadow: inset 0px 0px 25px rgba(23,246,251,1.0);
  animation: spinBG3 8s infinite linear;
  @media screen and (max-width: 500px) {
    width: 325px;
    height: 325px;
    margin-top: -150px;
  }
}

#break-countdown {
  margin-top: -200px;
  font-size: 70px;
  margin-top: -100px;
}

#clock-background {
  background: url(sunset-bg.png);
  opacity: 0.8;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  position: absolute;
  z-index: -1;
  background-size: cover;
  background-repeat: no-repeat;
}

.footer-link {
  color: rgb(255,225,213);

}

#footer  {
  position: absolute;
  left: 0;
  bottom: 0;
  width: 100%;
  color: white;
  text-align: center;
  margin-bottom: 10px;
}
/*-------------------------------------------
 Animations
-------------------------------------------*/
@keyframes pulseGlow{
	0%  {text-shadow: 0px 0px 20px rgba(23,246,251, 0.75);}
	50% {text-shadow: 0px 0px 40px rgba(23,246,251, 0.5); }
	100%{text-shadow: 0px 0px 20px rgba(23,246,251, 0.75);}		
}

@keyframes pulseShadow{
	0%  {box-shadow: 0px 0px 30px rgba(23,246,251, 0.25);}
	50% {box-shadow: 0px 0px 30px rgba(23,246,251, 0.75);}
	100%{box-shadow: 0px 0px 30px rgba(23,246,251, 0.25);}		
}

@keyframes spinBG{
	0%  {transform: rotate(0deg);}
	100%{transform: rotate(360deg);}	
}

@keyframes spinBG2{
	0%{
		transform: rotate(360deg);
		box-shadow: 0px 0px 1px rgba(23,246,251, 0.5);
	}
	50%{
		transform: rotate(180deg);
		box-shadow: 0px 0px 20px rgba(23,246,251, 0.5);
	}
	100%{
		transform: rotate(0deg);
		box-shadow: 0px 0px 1px rgba(23,246,251, 0.5);
	}	
}

@keyframes spinBG3{
	0%{
		transform: rotate(180deg);
		box-shadow: 0px 0px 1px rgba(23,246,251, 0.1);
	}
	50%{
		transform: rotate(0deg);
	}
	100%{
		transform: rotate(-180deg);
	}	
}

@keyframes glow {
  from {
    text-shadow: 0 0 10px #fff, 0 0 20px #fff, 0 0 30px #908ab9, 0 0 40px #908ab9, 0 0 50px #908ab9, 0 0 60px #908ab9, 0 0 70px #908ab9;
  }
  to {
    text-shadow: 0 0 20px #fff, 0 0 30px #ff4da6, 0 0 40px #ff4da6, 0 0 50px #ff4da6, 0 0 60px #ff4da6, 0 0 70px #ff4da6, 0 0 80px #ff4da6;
  }
}

