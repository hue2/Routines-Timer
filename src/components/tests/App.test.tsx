import * as React from 'react';
import * as ReactDOM from 'react-dom';
import App from '../App';
import * as renderer from 'react-test-renderer';

describe('App tests', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<App />, div);
    });

    it('component matches snapshot', () => {
      const tree = renderer.create(<App />).toJSON();
      expect(tree).toMatchSnapshot();
    });
})