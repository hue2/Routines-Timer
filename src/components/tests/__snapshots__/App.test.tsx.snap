// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests component matches snapshot 1`] = `
<div
  id="main"
>
  <div
    id="clock-background"
  />
  <div
    className="App"
  >
    <div
      id="volume-wrapper"
    >
      <audio
        autoPlay={true}
        id="player"
        src=""
      />
      <div>
        <div
          id="volume-icon"
          onClick={[Function]}
        >
          <i
            className="fa fa-volume-up"
          />
        </div>
        <input
          disabled={false}
          id="volume-slider"
          max="1"
          min="0"
          name="volume"
          onChange={[Function]}
          step={0.1}
          type="range"
          value={0.5}
        />
      </div>
    </div>
    <div
      className="hidden"
    >
      <ul>
        <li />
        <li />
        <li />
        <li />
        <li />
      </ul>
    </div>
    <div>
      <div
        className="time"
        id="time-countdown"
      >
        <b>
          00:00:00
        </b>
      </div>
      <div
        className="hidden"
        id="break-countdown"
      >
        <h3>
          Break time!
        </h3>
        <b>
          00:00:00
        </b>
      </div>
      <div
        className="hidden"
        id="repeat-txt"
      >
        <p
          className="small-text"
        >
          Laps remaining: 
          0
        </p>
      </div>
    </div>
    <div
      className=""
      title="Click settings below to set time"
    >
      <button
        className="timer-btn disabled-btn "
        disabled={true}
        id="start-btn"
        onClick={[Function]}
      >
        <i
          className="fa fa-play"
        />
          Start
      </button>
    </div>
    <div
      className="hidden"
      id="button-groups"
    >
      <button
        className="timer-btn "
        id="reset-btn"
        onClick={[Function]}
      >
        <i
          className="fa fa-times-circle-o"
        />
          Reset
      </button>
      <button
        className="timer-btn "
        id="pause-btn"
        onClick={[Function]}
      >
        <i
          className="fa fa-pause-circle-o"
        />
          Pause
      </button>
      <button
        className="timer-btn "
        id="restart-btn"
        onClick={[Function]}
      >
        <i
          className="fa fa-refresh"
        />
          Restart
      </button>
    </div>
    <div
      className=" overlay close"
      id="myNav"
    >
      <a
        className="closebtn"
        href="javascript:void(0)"
        onClick={[Function]}
      >
        ×
      </a>
      <div
        className="overlay-content"
      >
        <div
          className="option-div"
        >
          <div
            className="option-label"
          >
            Time: 
          </div>
          <div
            className="input-time"
          >
            <input
              className="hour"
              data-state="time"
              id="hour"
              maxLength={3}
              min="1"
              onChange={[Function]}
              placeholder="hh"
              type="number"
              value=""
            />
            <label
              className="dot-separator"
            >
              :
            </label>
            <input
              className="minute"
              data-state="time"
              maxLength={3}
              min="1"
              onChange={[Function]}
              placeholder="mm"
              type="number"
              value=""
            />
            <label
              className="dot-separator"
            >
              :
            </label>
            <input
              className="second"
              data-state="time"
              maxLength={3}
              min="1"
              onChange={[Function]}
              placeholder="ss"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          className="option-div"
        >
          <div
            className="option-label"
          >
            Breaks: 
          </div>
          <div
            className="input-time"
          >
            <input
              className="minuteBreak"
              data-state="timeBreak"
              maxLength={3}
              min="1"
              onChange={[Function]}
              placeholder="mm"
              type="number"
              value=""
            />
            <label
              className="dot-separator"
            >
              :
            </label>
            <input
              className="secondBreak"
              data-state="timeBreak"
              maxLength={3}
              min="1"
              onChange={[Function]}
              placeholder="ss"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          className="option-div"
        >
          <div
            className="option-label"
          >
            Repeats: 
          </div>
          <div
            className="input-time"
          >
            <input
              className="repeat tempRepeat"
              data-state="repeats"
              maxLength={3}
              min="1"
              onChange={[Function]}
              type="number"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className=""
      data-testid="settings-btn"
      id="settings-btn"
      onClick={[Function]}
    >
      <i
        className="fa fa-gear"
      />
      Adjust Time
    </div>
  </div>
  <div
    id="footer"
  >
    Image by 
    <a
      className="footer-link"
      href="https://pixabay.com/users/moinzon-2433302/?utm_source=link-attribution&utm_medium=referral&utm_campaign=image&utm_content=1412683"
    >
      Michi S
    </a>
     from  
    <a
      className="footer-link"
      href="https://pixabay.com/?utm_source=link-attribution&utm_medium=referral&utm_campaign=image&utm_content=1412683"
    >
      Pixabay
    </a>
    <br />
    <br />
    See the project on 
    <a
      className="footer-link"
      href="https://github.com/hue2/Routines-Timer"
    >
      Github
    </a>
  </div>
</div>
`;
